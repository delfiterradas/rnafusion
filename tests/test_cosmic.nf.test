nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    profile "test_cosmic"
    tag "pipeline"
    tag "pipeline_rnafusion"
    tag "test_cosmic"

    test("test cosmic with fastp trim") {

        when {
            params {
                outdir               = "$outputDir"
                fastp_trim           = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("test cosmic no fastp trim") {

        when {
            params {
                outdir     = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("test cosmic no fastp trim build") {

        when {
            params {
                outdir           = "$outputDir"
                build_references = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            // Ignore files with timestamps in their names
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
