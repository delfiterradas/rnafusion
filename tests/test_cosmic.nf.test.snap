{
    "test cosmic no fastp trim build": {
        "content": [
            0,
            {
                "Workflow": {
                    "nf-core/rnafusion": "v4.0.0dev"
                }
            },
            [
                "pipeline_info",
                "pipeline_info/nf_core_rnafusion_software_mqc_versions.yml"
            ],
            [

            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T12:42:49.558429"
    },
    "test cosmic no fastp trim": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ],
            [

            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T15:09:53.130295"
    },
    "test cosmic with fastp trim": {
        "content": [
            0,
            null,
            [
                "pipeline_info"
            ],
            [

            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T15:09:29.277399"
    }
}
